bindings/rascal/utils/cur.py

```diff
@@ -120,86 +135,96 @@ def fit_transform(self, managers):
-                managers, selected_ids_by_sp, strides_by_sp, map_by_manager)
+        return self
 
-            # build the pseudo points
-            sparse_points = SparsePoints(self._representation)
-            sparse_points.extend(managers, self.selected_ids)
+    def transform(self, managers, n_select=None):
+        if n_select is None:
+            n_select = self.Nselect
 
-            return sparse_points
-        else:
-            raise NotImplementedError("method: {}".format(self.act_on))
-
-    def get_index_mappings_sample_per_species(self, managers):
-        # get various info from the structures about the center atom species and indexing
-        sps = list(self.Nselect.keys())
-        types = []
-        strides_by_sp = {sp: [0] for sp in sps}
-        global_counter = {sp: 0 for sp in sps}
-        indices_by_sp = {sp: [] for sp in sps}
-        map_by_manager = [{} for ii in range(len(managers))]
-        for i_man, man in enumerate(managers):
-            counter = {sp: 0 for sp in sps}
-            for i_at, at in enumerate(man):
-                types.append(at.atom_type)
-                if at.atom_type in sps:
-                    map_by_manager[i_man][global_counter[at.atom_type]] = i_at
-                    counter[at.atom_type] += 1
-                    global_counter[at.atom_type] += 1
-                else:
-                    raise ValueError('Atom type {} has not been specified in fselect: {}'.format(
-                        at.atom_type, self.Nselect))
-            for sp in sps:
-                strides_by_sp[sp].append(counter[sp])
-
-        for sp in sps:
-            strides_by_sp[sp] = np.cumsum(strides_by_sp[sp])
-
-        for ii, sp in enumerate(types):
-            indices_by_sp[sp].append(ii)
-
-        return strides_by_sp, global_counter, map_by_manager, indices_by_sp
-
-    def convert_selected_global_index2rascal_sample_per_species(self, managers,
-                            selected_ids_by_sp, strides_by_sp, map_by_manager):
-        # convert selected center indexing into the rascal format
-        selected_ids = [[] for ii in range(len(managers))]
-        sps = list(self.Nselect.keys())
-        i_manager = {sp: 0 for sp in sps}
-        for sp in sps:
-            for idx in selected_ids_by_sp[sp]:
-                carry_on = True
-                while carry_on:
-                    if (idx >= strides_by_sp[sp][i_manager[sp]] and
-                                    idx < strides_by_sp[sp][i_manager[sp] + 1]):
-                        selected_ids[i_manager[sp]].append(
-                            map_by_manager[i_manager[sp]][idx])
-                        carry_on = False
-                    else:
-                        i_manager[sp] += 1
-        for ii in range(len(selected_ids)):
-            selected_ids[ii] = list(np.sort(selected_ids[ii]))
-        return selected_ids
+        if self.act_on in ['sample per species']:
```

Comment:
Yes that's what it does for every element of the list. Is there a problem ?

Reply:
Yes, the list has one element, so the `==` form is simpler. It took me several tries to figure out what this does; with the `==` form it would have been instantly obvious.